### PR TITLE
Indirect - Replace old output plotting widgets

### LINKED ISSUE #26029: Indirect Reduction - Create Widget for plotting output data
### Description
A widget should be created which will can be used to plot the output data on data Reduction and other indirect interfaces. This is a maintenance task because the code that currently exists needs to be replaced so the output plotting uses the same widget.
This widget needs:
- A `Plot Contour` button which can be hidden if necessary. This button will open the SliceViewer.
- A `Plot Spectra` button which will plot the spectra specified by the indices in a line edit.
- A line edit which allows you to enter the indices within a workspace which you want to plot e.g. `0-5, 8, 11-15`. Default should be a small number of spectra, maybe just `0`.

This widget can be used across all of the reduction interfaces, and perhaps in other Indirect interfaces too.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR replaces the old Indirect output plotting code with a refactored widget that can be used across most of the Indirect Interfaces. It has only been implemented for the Data Reduction interface initially for this PR.

The new widget has a few usability improvements:
- The workspace currently selected in the output options is now visible to the user (shown in a combobox).
- The widget allows you to select a list of spectra for plotting (e.g. 0-2,4,6-8), rather than just a single spectrum (like the previous widgets did).
- The same widget is used across all the tabs in Data Reduction, so consistency has also improved.
- Deleting or replacing an output workspace in the ADS will no longer cause problems.

Behind the scenes, the widget is coded in an MVP pattern and has its own unit tests. Release notes have been added.

**To test:**
Test on MantidPlot and Workbench
1. `Interfaces`->`Indirect`->`Data Reduction`->`ISIS Energy Transfer`
2. Select `IRIS` instrument.
3 Load files `26173-26176`
4. Select `Individual` detector grouping.
5. Click `Run` and wait.
6. The plotting options should show 4 workspaces in the combobox. 
7. Select a workspace and enter `0-2,4` in the neighbouring box. Click `Plot Spectra`. This should plot these indices.
8. Then, select a workspace and click the down arrow on the `Plot Spectra` button, and then click `Plot Contour`. This will produce a contour plot of the selected workspace. 

9. Go to `ISIS Calibration`
10. Load `26173`
11. Tick `Create RES`
12. Click `Run`
13. Choose the workspace ending in `_calib`. Click the down arrow on the `Plot Spectra` button, and then click `Plot Bin`. This will produce a bin plot.
14. Choose the workspace ending in `_res`. Click `Plot Spectra`. This produces a spectrum plot.

Fixes #26029

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by BassoLorenzo — DISMISSED** [review]
Works as advertised, no problems were found.

File: qt/scientific_interfaces/Indirect/ISISCalibration.cpp [hunk 1]
```diff
@@ -42,6 +42,8 @@ namespace CustomInterfaces {
 ISISCalibration::ISISCalibration(IndirectDataReduction *idrUI, QWidget *parent)
     : IndirectDataReductionTab(idrUI, parent), m_lastCalPlotFilename("") {
   m_uiForm.setupUi(parent);
+  setOutputPlotOptionsPresenter(std::make_unique<IndirectPlotOptionsPresenter>(
+      std::move(m_uiForm.ipoPlotOptions), this, PlotWidget::SpectraBin));
```
Comment (gemmaguest) [inline qt/scientific_interfaces/Indirect/ISISCalibration.cpp]: I don't think we want to transfer ownership of this to the presenter so this should probably be passed as a raw pointer (similar to uses of `m_uiForm->fitDataView`). Similarly for other uses.

**Review by gemmaguest — CHANGES_REQUESTED** [review]
Just one minor thing before I'm happy to merge

File: qt/scientific_interfaces/Indirect/test/IndirectPlotOptionsModelTest.h [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+  }
+
+  static void destroySuite(IndirectPlotOptionsModelTest *suite) {
+    delete suite;
+  }
+
+  void setUp() override {
+    m_plotter = std::make_unique<NiceMock<MockIndirectPlotter>>();
+    m_model =
+        std::make_unique<IndirectPlotOptionsModel>(std::move(m_plotter.get()));
```
Comment (gemmaguest) [inline qt/scientific_interfaces/Indirect/test/IndirectPlotOptionsModelTest.h]: I'm not sure the move is necessary here and I think it ends up being owned both locally and by the model? Would it be better to just create it as a raw pointer here if the model takes ownership?

**Review by gemmaguest — CHANGES_REQUESTED** [review]
Just spotted another use of `move` which looks unnecessary

**Comment by gemmaguest** [comment]
@LolloB Could you quickly re-test this to make sure it still works after the recent changes? Then I'll merge it.

**Review by BassoLorenzo — APPROVED** [review]
Works well on MantidPlot and Workbench for Ubuntu 18.04

**Review by gemmaguest — APPROVED** [review]

**Pull request merged**